models/repo/repo.go

```diff
@@ -765,35 +765,32 @@ func CountRepositories(ctx context.Context, opts CountRepositoryOptions) (int64,
 	return count, nil
 }
 
-// StatsCorrectNumClosed update repository's issue related numbers
-func StatsCorrectNumClosed(ctx context.Context, id int64, isPull bool, field string) error {
-	_, err := db.Exec(ctx, "UPDATE `repository` SET "+field+"=(SELECT COUNT(*) FROM `issue` WHERE repo_id=? AND is_closed=? AND is_pull=?) WHERE id=?", id, true, isPull, id)
-	return err
-}
-
-// UpdateRepoIssueNumbers update repository issue numbers
+// UpdateRepoIssueNumbers updates one of a repositories amount of (open|closed) (issues|PRs) with the current count
 func UpdateRepoIssueNumbers(ctx context.Context, repoID int64, isPull, isClosed bool) error {
-	e := db.GetEngine(ctx)
+	field := "num_"
+	if isClosed {
+		field += "closed_"
+	}
 	if isPull {
-		if _, err := e.ID(repoID).Decr("num_pulls").Update(new(Repository)); err != nil {
-			return err
-		}
-		if isClosed {
-			if _, err := e.ID(repoID).Decr("num_closed_pulls").Update(new(Repository)); err != nil {
-				return err
-			}
-		}
+		field += "pulls"
 	} else {
-		if _, err := e.ID(repoID).Decr("num_issues").Update(new(Repository)); err != nil {
-			return err
-		}
-		if isClosed {
-			if _, err := e.ID(repoID).Decr("num_closed_issues").Update(new(Repository)); err != nil {
-				return err
-			}
-		}
+		field += "issues"
 	}
-	return nil
+
+	var cond builder.Cond = builder.Eq{
+		"repo_id": repoID,
+		"is_pull": isPull,
+	}
+	if isClosed {
+		cond = cond.And(builder.Eq{"is_closed": isClosed})
+	}
+	subQuery := builder.Select("count(*)").From("issue").Where(cond)
+
+	query := builder.Update(builder.Eq{field: subQuery}).
```

Comment:
is this the intended usage? to use builder eq for an update statement?!?

Reply:
What is going on here is that the builder is misused to set the field to update dynamically.
But yeah, maybe we should document this line as I've also struggled with it, and it certainly isn't as apparent as is typical for Go code.

Reply:
for my understanding `builder.Eq{field: subQuery}` is equivalten to go-code `fmt.Sprintf("%s = %s", field, subQuery)

Reply:
Here Update means Set and Update

Reply:
@lunny can you add some code comment to explain for others :)

Reply:
Done. And in fact, this is not the first place we use `builder.Update` in codebase.